### PR TITLE
Add support for the Actinius Icarus board (nRF9160)

### PR DESCRIPTION
**Pull request by alextsam**
With this PR we are adding support for the Icarus board (www.actinius.com/icarus) which is based on the nRF9160.

### CODE REVIEW

**Comment by SebastianBoe** [comment]
Hi, thank you for this contribution.

We are glad to see the nRF91 on this board and would like to see it integrated with NCS. But we have decided that the support should be added upstream first, and then (e.g. through patching boilerplate.cmake) supported in NCS afterwards.

Would you kindly post this PR upstream instead?

**Comment by alextsam** [comment]
Hi @SebastianBoe, thank you for your message. I will then submit the PR right now to upstream without the boilerplate patch and once it is merged I will open a new PR here for the patch, is that right?
Thanks.

**Comment by nvlsianpu** [comment]
@alextsam 
> Hi @SebastianBoe, thank you for your message. I will then submit the PR right now to upstream without the boilerplate patch and once it is merged I will open a new PR here for the patch, is that right?
Thanks.

Yes without boilerplate patch as it is relevant for NCS only.
We will integrate zephyr changes in regular upmerge so no need to open another PR - you can just make a PR with boilerplate patch here, which will be expected to be merged after the upmerge.

**Comment by SebastianBoe** [comment]
That's right, thank you.

File: boards/arm/actinius_icarus/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+Building Secure/Non-Secure Zephyr applications
+==============================================
+
+The process requires the following steps:
+
+1. Build the Secure Zephyr application using ``-DBOARD=actinius_icarus`` and
+   ``CONFIG_TRUSTED_EXECUTION_SECURE=y`` in the the application project configuration file.
+2. Build the Non-Secure Zephyr application using ``-DBOARD=actinius_icarus_ns``.
+3. Merge the two binaries together.
```
Comment (hakonfam) [inline boards/arm/actinius_icarus/doc/index.rst]: Aren't the files merged automatically? If I build hello_world sample with `actinios_icarus_ns` board selected, I get `build/zephyr/merged.hex` generated by the build system automatically.

**Comment by SebastianBoe** [comment]
Closing PR to avoid duplicate review.

**Pull request closed without merging**